### PR TITLE
Wip solution

### PR DESCRIPTION
**Pull request by azat-co**
CC @AnnJohn @vicfriedman

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,41 @@
...
+
+## Objectives
+
+1. Implement a folder as a proxy for modules
+1. Create files in a folder, files with exports
+1. Import files/modules from another file via folder "proxy"
+
+## Introduction
+
+In this lab, you'll learn how to proxy folders or in other words how to create an abstraction layer which will allow you to group files using the Node feature of importing folders.
```
Comment (victhevenot) [inline README.md]: Reframe this - What is the "domain" of the lab, ie. what kind of files are they working with, what kind of project is this contributing to?

File: README.md [hunk 2]
```diff
@@ -1 +1,41 @@
...
+
+## Instructions
+
+1. Create a proxy file `index.js` in the `color-logs` folder
+2. Import all the files in the `color-logs` folder in the `index.js` (except `index.js obviously)
```
Comment (victhevenot) [inline README.md]: what kind of code in contained in `color-logs`?

File: README.md (hunk 1 above)
Comment (victhevenot) [inline README.md]: definitely give an overview of what the code in `color-logs` does.

File: test/index.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+var test = require('tape'),
```
Comment (victhevenot) [inline test/index.js]: refactor to mocha + chai / make tests readable

**Comment by azat-co** [comment]
Updated CC @vicfriedman @AnnJohn

**Comment by victhevenot** [comment]
going to @pletcher next, but a few comments for @azat-co:
- When students will be using new material in a lab that hasn't been addressed in a previous readme, they tend to think it's a mistake on our part. make sure to explicitly tell students we purposefully haven't taught them everything they need to know to complete this lab - they will need to use the resources to do some research on their own.
- on line 74, add the header `Resources`
- The tests don't check the console.log output of the text - @pletcher thoughts there?
- I would more clearly explain the `%s` when setting and clearing the color - students have never been explicitly taught interpolation with `%s` and its not totally clear from the contents of the readme now how thats happening

File: README.md [hunk 4]
```diff
@@ -1 +1,78 @@
...
+
+Code organization is important. If you want to really master Node and impress you co-workers, you need to know how to import folders. This powerful pattern is sometimes called a proxy for modules. This plugin pattern allows you to write a line of code to import a directory, which also recursively imports everything inside that directory. 
+
+Imagine you're working on a server logger. You must implement a series of modules which display your app logs in different colors. (It can be later used in a CLI game development too!).
+
+## Instructions
+
+1. Create a proxy file `index.js` in the `color-logs` directory.
+2. Import all the other JS files in the `color-logs` directory in the `index.js` (except `index.js` obviously)
+3. Export the imported file as an object. Follow the  `module.exports = {...}` pattern with filenames as values of properties (see extra info if this is too confusing)
```
Comment (pletcher) [inline README.md:19]: We want to export _all_ of the imported files as an object.

File: README.md [hunk 5]
```diff
@@ -1 +1,78 @@
...
+4. Create `main.js` in the root directory of this project and import the `color-logs` folder.
+5. Print two lines: "Access granted" in the green color and "Permission denied" in red.
+6. Bonus: refactor `index.js` to process an unlimited number of files from the `color-logs` folder with `fs` **without** using any extra npm modules like `require-dir`. You can use `readdirSync()` to read the list of files in the folder. Note: Don't forget to check that the files have `.js` extension and remember to convert any weird names (e.g., `deep-blue.js`) to proper JavaScript/Node names with camelCase (`deepBlue`).
```
Comment (pletcher) [inline README.md:22]: Strictly speaking, the weird names would be okay. Maybe offer an example about why it's a good idea to convert them to camelCase (e.g., calling `colors['deep-blue'](text)`, while possible, is a little funky).

I'm not sure that we need to include the phrase "proper JavaScript/Node names" -- we can just show students why they might prefer camelCase.

File: test/index.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
+var expect = require('chai').expect
+var cp = require('child_process')
+
+describe('main.js', function () {
+  it('must print green, red and yellow logs', function(done){
+    child = cp.exec('node main', function (error, output, errorOutput) {
```
Comment (pletcher) [inline test/index.js:6]: I think it's overkill to spawn a child process here. Why not test the module directly?

**Comment by pletcher** [comment]
I like this lab, but the conclusion leaves me wanting more. Could we get the students to build a colorizing module that they can import into their projects? No need to publish it or anything (there are enough of those), obviously, but just something that they can use to colorize text. All of the basic tools are there, and that gives them a chance to explore the module system a bit more and to provide a broader context than simply printing out two lines.

**Pull request closed without merging**